src/app.ts

```diff
@@ -0,0 +1,175 @@
+
+    private readonly continousIntegrationConfigurationJsonFileName: PathLike = '.laminas-ci.json';
+
+    public constructor(
+        private readonly action: Action,
+        private readonly logger: Logger
+    ) {}
+
+    public sanityChecksPassing(): boolean {
+        let schemaValidationResult = this.projectContainsValidJsonConfiguration(
+            'composer.json.schema',
+            this.composerJsonFileName
+        );
+
+        if (!schemaValidationResult.valid) {
+            return this.failWithLoggedJsonSchemaValidationErrors(this.composerJsonFileName, schemaValidationResult);
+        }
+
+        schemaValidationResult = this.projectContainsValidJsonConfiguration(
+            'laminas-ci.schema.json',
+            this.continousIntegrationConfigurationJsonFileName
+        );
+
+        if (!schemaValidationResult.valid) {
+            return this.failWithLoggedJsonSchemaValidationErrors(
+                this.continousIntegrationConfigurationJsonFileName,
+                schemaValidationResult
+            );
+        }
+
+        return true;
+    }
+
+    private validateJsonSchema(
+        pathToJsonToValidate: PathLike,
+        pathToSchemaJsonForValidation: PathLike
+    ): ValidationResult {
+        this.logger.debug(`Running ${pathToJsonToValidate} linting using ${pathToSchemaJsonForValidation}.`);
+
+        const jsonSchema = parseJsonFile(pathToSchemaJsonForValidation, false);
+        const jsonSchemaValidator = new Validator(jsonSchema);
+        const validationResult = jsonSchemaValidator.validate(parseJsonFile(pathToJsonToValidate, false));
+
+        if (validationResult.valid) {
+            this.logger.debug(`${pathToJsonToValidate} schema validation passed.`);
+        }
+
+        return validationResult;
+    }
+
+    createJobs(
+        appConfig: Config
+    ): [JobForMatrix, ...JobForMatrix[]] {
+        const config = parseJsonFile(this.continousIntegrationConfigurationJsonFileName, true) as ConfigurationFromFile;
+        const tools = createTools(appConfig);
+
+        this.logger.debug(`Tools detected: ${JSON.stringify(tools, null, SPACES_TO_INDENT_JSON)}`);
+        const jobs: [Job, ...Job[]] = (new DefaultJobCreator(tools))
+            .createJobs(config, appConfig);
+
+        this.logger.debug(`Jobs created: ${JSON.stringify(jobs, null, SPACES_TO_INDENT_JSON)}`);
+
+        return jobs.map((job) => createJobForMatrixFromJob(job)) as [JobForMatrix, ...JobForMatrix[]];
+    }
+
+    createConfiguration(filesFromDiff: string[]): Config {
+        return createConfigObject(
+            this.checkRequirements(filesFromDiff),
+            this.composerJsonFileName,
+            this.composerLockJsonFileName,
+            this.continousIntegrationConfigurationJsonFileName
+        );
+    }
+
+    private projectContainsValidJsonConfiguration(schemaName: PathLike, fileName: PathLike): ValidationResult {
+        const schemaPath = `${this.action.getApplicationDirectory()}/${schemaName}`;
```

Comment:
Very convoluted way to concatenate two strings, but okay O_o

Reply:
This is pretty typical in TS

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals